src/rpc/evo.cpp

```diff
@@ -1267,25 +1267,39 @@ static bool CheckWalletOwnsScript(CWallet* pwallet, const CScript& script) {
 }
 #endif
 
-static UniValue BuildDMNListEntry(CWallet* pwallet, const CDeterministicMN& dmn, bool detailed)
+static UniValue BuildDMNListEntry(CWallet* pwallet, const CDeterministicMN& dmn, bool detailed, const CBlockIndex* pindex = nullptr)
 {
     if (!detailed) {
         return dmn.proTxHash.ToString();
     }
 
     UniValue o = dmn.ToJson();
 
+    CTransactionRef collateralTx{nullptr};
     int confirmations = GetUTXOConfirmations(dmn.collateralOutpoint);
+
+    if (pindex != nullptr) {
+        if (confirmations > -1) {
+            confirmations -= WITH_LOCK(cs_main, return ::ChainActive().Height()) - pindex->nHeight;
+        } else {
+            uint256 minedBlockHash;
+            collateralTx = GetTransaction(/* pindex */ nullptr, /* mempool */ nullptr, dmn.collateralOutpoint.hash, Params().GetConsensus(), minedBlockHash);
```

Comment:
should you specify here proper pindex, so, it can work if there's no tx-index available on node?

Reply:
but we don't know it at this point 🤷‍♂️ `pindex` in `BuildDMNListEntry` is the block index of the dmn list and what `GetTransaction` needs is the block index of the collateral tx (we obtain it later as `pindexMined` using `minedBlockHash` returned by `GetTransaction`).